### PR TITLE
Add django 1.8 support

### PR DESCRIPTION
**Pull request by GertBurger**
This adds Django 1.7/1.8 support but changes the minimum Django version to 1.7

The original south migrations are still there and South 1.x will find them but from this point onwards the Django migrations should be used.

### CODE REVIEW

**Comment by czue** [comment]
@emord fyi for django 1.7 upgrade.

cc @kennknowles - on the off chance you are interested in taking a look

**Comment by emord** [comment]
Looks good to me, but I'm not really familiar with the library. Probably want to bump the version though. `0.1.0` ?

**Comment by emord** [comment]
btw holding off on merging this until we upgrade HQ. any reason we use this as a submodule instead of a package?

**Comment by czue** [comment]
I think we could convert it to a package. seems stable enough at this point.

**Comment by emord** [comment]
Default fields will need to be converted away from mutable objects: https://docs.djangoproject.com/en/1.8/ref/models/fields/#django.db.models.Field.default

https://github.com/dimagi/django-prbac/pull/13/files#diff-7abae2b3a51ca11883307e228813eb09L124

File: django_prbac/migrations/0001_initial.py [hunk 1]
```diff
@@ -1,57 +1,56 @@
...
+            bases=(django_prbac.models.ValidatingModel, models.Model),
+        ),
+        migrations.CreateModel(
+            name='Role',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
+                ('slug', models.CharField(help_text='The formal slug for this role, which should be unique', unique=True, max_length=256)),
+                ('name', models.CharField(help_text='The friendly name for this role to present to users; this need not be unique.', max_length=256)),
+                ('description', models.TextField(default='', help_text='A long-form description of the intended semantics of this role.', blank=True)),
+                ('parameters', django_prbac.fields.StringSetField(default=list, help_text='A set of strings which are the parameters for this role. Entered as a JSON list.', blank=True)),
```
Comment (emord) [inline django_prbac/migrations/0001_initial.py]: This should be `default=set`

File: django_prbac/models.py [hunk 2]
```diff
@@ -58,7 +56,7 @@ class Role(ValidatingModel, models.Model):
     parameters = StringSetField(
         help_text='A set of strings which are the parameters for this role. Entered as a JSON list.',
         blank=True,
-        default=[],
+        default=list,
```
Comment (emord) [inline django_prbac/models.py]: `set`

**Comment by GertBurger** [comment]
I did find it odd that the previous value was [] for a 'set' field but it functioned correctly in anycase so I left it. Fixed now.

**Comment by emord** [comment]
Agreed that was strange. Not sure how that worked before. Can you also merge master? (not actually possible for me to do it on your fork)

**Comment by GertBurger** [comment]
I just noticed that 2 tests are failing, I haven't checked why or if they were failing beforehand. Ill get to that tomorrow.

**Comment by emord** [comment]
Which tests are you talking about? The tests are passing on travis.

Thanks a bunch for upgrading this.

**Comment by GertBurger** [comment]
I'm trying to run them locally via nose. I'll check out travis's
configuration to see how it runs tests and take it from there.

**Comment by emord** [comment]
Thanks @GertBurger !

**Pull request merged**